### PR TITLE
Blackpowder Reloads and Reloaded Ammo Disassembly

### LINKED ISSUE #30200: Allow disassembly on reloaded ammo
**Is your feature request related to a problem? Please describe.**  
In this game, you can reload ammo from various materials pretty easily. However, the reloaded ammo cannot be disassembled even thought it would be pretty easy to disassemble.

**Describe the solution you'd like**  
Make them disassemblable and add the materials gained from it.

**Describe alternatives you've considered**  
If you meant for the bullets to never be able to be disassembled, add a description that says that.

### PR DESCRIPTION
**Pull request by Greevv**
#### Summary
SUMMARY: Content "Adds black powder reloads for compatible calibers and allows for the disassembly of reloaded ammo."

#### Purpose of change
This PR allows the creation of black powder reloads for cartridges with real-world black powder loads. The cartridges affected are all revolver/single shot cartridges that are not intended for usage in semi-automatics.

Also fixes #30200 by allowing for the disassembly of reloaded ammunition at the cost of not allowing the creation of reloaded ammo with other metals, which never really made sense anyways considering that bullet weight is VERY important when calculating load data and having too heavy/too light of a bullet could cause dire consequences.

#### Describe the solution
- Created a few different black powder cartridge variants and added recipes for them.
- Removed metals other than lead from bullet crafting.
- Added code in requirements.cpp to handle the disassembly of reloaded ammo, fire and hand press are removed from the requirements, but a bullet puller is needed.

#### Additional context
This chart is where I got most of my load data. For those unaware, Goex is a very large producer of black powder and has been for years.
https://goexpowder.com/wp-content/uploads/2018/06/cartridge-rifle.pdf

### CODE REVIEW

File: data/json/items/ammo/38.json [hunk 1]
```diff
@@ -65,5 +65,15 @@
     "proportional": { "price": 0.7, "damage": 0.9, "dispersion": 1.1 },
     "extend": { "effects": [ "RECYCLED" ] },
     "delete": { "effects": [ "NEVER_MISFIRES" ] }
+  },
+  {
+    "id": "bp_38_special",
+    "copy-from": "38_special",
+    "type": "AMMO",
+    "name": "blackpowder .38 Special",
+	"description": "This .38 Special cartridge is loaded with a 200gr cast lead hollowpoint and 23gr of blackpowder. Much slower than regular .38s, but tremendous stopping power.",
```
Comment (Night-Pryanik) [inline data/json/items/ammo/38.json]:
```suggestion
	"description": "This .38 Special cartridge is loaded with a 200gr cast lead hollowpoint and 23gr of blackpowder. Much slower than regular .38s, but has tremendous stopping power.",
```

File: data/json/recipes/ammo/shot.json [hunk 2]
```diff
@@ -11,8 +11,9 @@
     "batch_time_factors": [ 60, 5 ],
     "book_learn": [ [ "recipe_bullets", 2 ], [ "manual_shotgun", 2 ] ],
     "charges": 1,
+	"reversible": true,
```
Comment (Night-Pryanik) [inline data/json/recipes/ammo/shot.json]: Tabs, here and below.

File: data/json/items/ammo/38.json (hunk 1 above)
Comment (Greevv) [reply data/json/items/ammo/38.json]: I disagree. It's a sentence fragment and I don't feel that adding "has" would improve clarity at all.

File: data/json/recipes/ammo/shot.json (hunk 2 above)
Comment (Greevv) [reply data/json/recipes/ammo/shot.json]: Thanks, I will fix that in the morning.

**Comment by RarkGrames** [comment]
Is blackpowder loaded ammo generates excessive amount of smoke when fired? If so I would consider adding the MUZZLE_SMOKE effect to them, found on the paper cartridges. Or that amount of smoke is only applicable for the paper cartridges?

**Comment by Greevv** [comment]
IIRC I tried that and it didn't work, but I was planning on implementing
that one way or another. I'll give it another look this morning.

File: data/json/items/ammo/38.json (hunk 1 above)
Comment (anothersimulacrum) [reply data/json/items/ammo/38.json]: Even if it doesn't improve clarity, adding has makes it flow better (maybe `it has` would be better).

**Comment by Greevv** [comment]
I'll tweak it. Thanks for the recommendation.

**Comment by RarkGrames** [comment]
> 
> IIRC I tried that and it didn't work, but I was planning on implementing that one way or another. I'll give it another look this morning.

I tried it and it worked for me. A lot of the ammo items use copy_from so only the base one has to have it.

File: data/json/items/ammo/40mm.json [hunk 3]
```diff
@@ -107,5 +107,15 @@
     "description": "A 40mm grenade with a teargas load.  It will burst in a cloud of highly incapacitating gas.",
     "drop": "gasbomb_act",
     "extend": { "effects": [ "NO_EMBED" ] }
+  },
+  {
+    "id": "40mm_slug",
```
Comment (kevingranade) [inline data/json/items/ammo/40mm.json]: Have a source for this? 40mm rounds are notorious for requiring a complex propellant mix to manage recoil, I would only imagine a slug is strictly worse.

**Comment by Greevv** [comment]
I agree, that's why I gave it a much higher recoil value.

File: data/json/items/ammo/40mm.json (hunk 3 above)
Comment (Greevv) [reply data/json/items/ammo/40mm.json]: I agree, that's why I gave it a much higher recoil value.

Comment (Greevv) [reply data/json/items/ammo/40mm.json]: If you think the recoil is still insufficient I can raise it.

Comment (kevingranade) [reply data/json/items/ammo/40mm.json]: On second thought, you can always sabot and shape your slug to reduce weight, so the question is how much slug and how much powder you can put in there.
The problem with trying to do reloads of the standard rounds is that your payload is fixed, but that doesn't apply here.

I'd still like to figure out what kind of load we're talking about here.

Comment (kevingranade) [reply data/json/items/ammo/40mm.json]: After some offline discussion I *think* we reached a consensus that:
Damage is fine, this is a monster round.
Recoil might be ok to drop a bit.
The recipe needs something to represent a sabot.
Probably drop the lead and gunpowder in the recipe.

**Comment by Greevv** [comment]
Can do, will try to get that done tomorrow.

**Pull request merged**